src/index.ts

```diff
@@ -74,7 +77,16 @@ export function refresh(): Promise<Types.NetInfoState> {
   if (!_state) {
     _state = createState();
   }
-  return _state._fetchCurrentState();
+
+  if (isRequestInProgress) {
+    return _state.latest(); // Return the latest state if a request is already in progress
```

Comment:
🤔 

If I understand this correctly, when multiple simultaneous requests are requested, only the first request will wait and get the correct / current state, all the subsequent ones (including the last one, which I presume will update the UI) will get incorrect / stale state. Is that correct?

Is there a way to have all the requests queue and wait on the first request to finish, then return the fresh information to all of them? May be possible with the creation of some Promises in an array or similar ?

Reply:
You are right, this solution may return an invalid state for subsequent requests. I modified it by adding a request queue, so all requests are queued while the first one is in progress. Once the first request completes, all queued requests are resolved with the updated state.